Makefile

```diff
@@ -0,0 +1,74 @@
+PREFIX ?= /usr/local
+
+.PHONY: release debug profile all clean install
+OPTLEVEL ?= -O3
+
+LIBS += sdl2 libpng libjpeg gl glew openal mad
+CXXFLAGS += -std=c++11 -Wall
+CXXFLAGS += $(shell pkg-config $(LIBS) --cflags)
+LDFLAGS += $(shell pkg-config $(LIBS) --libs) -lpthread
+
+PROG = endless-sky
+PROG_DEBUG = $(PROG)-debug
+PROG_PROF = $(PROG)-profile
+DIR_SRC = source
+DIR_BUILD = build
+DIR_RELEASE = $(DIR_BUILD)/release
+DIR_DEBUG = $(DIR_BUILD)/debug
+DIR_PROF = $(DIR_BUILD)/profile
+
+SRCS = $(wildcard $(DIR_SRC)/*.cpp)
+OBJS = $(patsubst $(DIR_SRC)/%.cpp,$(DIR_RELEASE)/%.o,$(SRCS))
+OBJS_DEBUG = $(patsubst $(DIR_SRC)/%.cpp,$(DIR_DEBUG)/%.o,$(SRCS))
+OBJS_PROF = $(patsubst $(DIR_SRC)/%.cpp,$(DIR_PROF)/%.o,$(SRCS))
+
+$(info mkdir -p $(DIR_RELEASE) $(DIR_DEBUG) $(DIR_PROF))
+$(shell mkdir -p $(DIR_RELEASE) $(DIR_DEBUG) $(DIR_PROF))
+
+$(DIR_RELEASE)/%.o: $(DIR_SRC)/%.cpp
+	$(CXX) $(OPTLEVEL) $(CXXFLAGS) -o $@ -c $<
+
+release: $(OBJS)
+	$(CXX) $(OPTLEVEL) $(CXXFLAGS) -o $(PROG) $^ $(LDFLAGS)
+
+$(DIR_DEBUG)/%.o: $(DIR_SRC)/%.cpp
```

Comment:
This won't work right when just a header file changes. (Not a problem with Scons, which seems to do dependency discovery and rebuilds only the minimum set of objects as necessary).

It's possible to do this correctly in Make (see gcc's `-M` options), but it's probably simpler to stick to a language-aware build system (like Scons).